### DISCUSSION TITLE
Lesson 6:TypeError: constructor requires no arguments

### DISCUSSION THREAD

**Discussion by ghost** [smartcontractkit/full-blockchain-solidity-course-py#819] [Q&A] [answered]
BrownieFundMeProject is the active project.

Launching 'ganache-cli --port 8545 --gasLimit 12000000 --accounts 10 --hardfork istanbul --mnemonic brownie'...

Running 'scripts/deploy.py::main'...
The active network is development
Deploying mocks...
Transaction sent: 0xb5a51c028dc765b846793bd11c254828027af45011d807608e0b487bea33d12e
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 0
  MockV3Aggregator.constructor confirmed   Block: 1   Gas used: 430695 (3.59%)
  MockV3Aggregator deployed at: 0x3194cBDC3dbcd3E11a07892e7bA5c3394048Cc87

Mocks deployed!
  File "brownie/_cli/run.py", line 49, in main
    return_value, frame = run(
  File "brownie/project/scripts.py", line 103, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File "./scripts/deploy.py", line 32, in main
Matts-MacBook-Air:brownie_fund_me mattwilson$ brownie run scripts/deploy.py
Brownie v1.16.4 - Python development framework for Ethereum
Matts-MacBook-Air:brownie_fund_me mattwilson$ brownie compile
Brownie v1.16.4 - Python development framework for Ethereum
Matts-MacBook-Air:brownie_fund_me mattwilson$ brownie run scripts/deploy.py
Brownie v1.16.4 - Python development framework for Ethereum
Matts-MacBook-Air:brownie_fund_me mattwilson$ brownie compiled
Brownie v1.16.4 - Python development framework for Ethereum

Invalid command. Did you mean 'brownie compile'?
Matts-MacBook-Air:brownie_fund_me mattwilson$ brownie compile
Brownie v1.16.4 - Python development framework for Ethereum
Matts-MacBook-Air:brownie_fund_me mattwilson$ brownie compile
Brownie v1.16.4 - Python development framework for Ethereum

Project has been compiled. Build artifacts saved at /Users/mattwilson/demos/brownie_fund_me/build/contracts
Matts-MacBook-Air:brownie_fund_me mattwilson$ brownie run scripts/deploy.py
Brownie v1.16.4 - Python development framework for Ethereum

BrownieFundMeProject is the active project.

Launching 'ganache-cli --port 8545 --gasLimit 12000000 --accounts 10 --hardfork istanbul --mnemonic brownie'...

Running 'scripts/deploy.py::main'...
The active network is development
Deploying mocks...
Transaction sent: 0xb5a51c028dc765b846793bd11c254828027af45011d807608e0b487bea33d12e
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 0
  MockV3Aggregator.constructor confirmed   Block: 1   Gas used: 430695 (3.59%)
  MockV3Aggregator deployed at: 0x3194cBDC3dbcd3E11a07892e7bA5c3394048Cc87

Mocks deployed!
  File "brownie/_cli/run.py", line 49, in main
    return_value, frame = run(
  File "brownie/project/scripts.py", line 103, in run
    return_value = f_locals[method_name](*args, **kwargs)
  File "./scripts/deploy.py", line 32, in main
    deploy_fund_me()
  File "./scripts/deploy.py", line 22, in deploy_fund_me
    fund_me = FundMe.deploy(
  File "brownie/network/contract.py", line 600, in __call__
    return tx["from"].deploy(
  File "brownie/network/account.py", line 507, in deploy
    data = contract.deploy.encode_input(*args)
  File "brownie/network/contract.py", line 629, in encode_input
    data = format_input(self.abi, args)
  File "brownie/convert/normalize.py", line 15, in format_input
    raise TypeError(f"{abi['name']} requires no arguments")
TypeError: constructor requires no arguments
Terminating local RPC client...
Matts-MacBook-Air:brownie_fund_me mattwilson$ 

i am currently getting this error on lesson 6 when i try to deploy the project. i have the constructor as this atm...

constructor() public {
        owner = msg.sender;
    }

because i tried this one... 

 constructor(address _priceFeed) public {
        priceFeed = AggregatorV3Interface(_priceFeed);
        owner = msg.sender;
    }

i deleted build folder and when i tried to compile again it said "priceFeed" undeclared identifier.

can i please get some help with this issue i am not too sure what to do. also i am quite new to coding and i find it really difficult of where to start when trying to solve an issue. is this a way to become better at identifying the source of a problem?

**Comment by cromewar** [comment, marked as answer]
Hello @mattwilsonbmx  this error should be happening on your deploy script, as you said the constructor requires no arguments, so when you use deploy are you sending anything else rather than the account? could you share your deploy script plz.

**Reply by ghost** [reply]
from brownie import network, config, FundMe, MockV3Aggregator

from scripts.helpful_scripts import get_account


```Python

def deploy_fund_me():
    account = get_account()

    if network.show_active() != "development":
        price_feed_address = config["networks"][network.show_active()][
            "eth_usd_price_feed"
        ]
    else:
        print(f"The active network is {network.show_active()}")
        print("Deploying mocks...")
        mock_aggregator = MockV3Aggregator.deploy(
            18, 2000000000000000000000, {"from": account}
        )
        price_feed_address = mock_aggregator.address
        print("Mocks deployed!")

    fund_me = FundMe.deploy(
        price_feed_address,
        {"from": account},
        publish_source=config["networks"][network.show_active()].get("verify"),
    )

    print(f"Contract deploy to {fund_me.address}")


def main():
    deploy_fund_me()
```

**Reply by cromewar** [reply]
As you can see you are sending  price_feed_address as parameter on:


```Python
    fund_me = FundMe.deploy(
        price_feed_address,
        {"from": account},
        publish_source=config["networks"][network.show_active()].get("verify"),
    )
```

but your solidity code does not have it
as you said you tried with:

```Solidity
constructor(address _priceFeed) public {
priceFeed = AggregatorV3Interface(_priceFeed);
owner = msg.sender;
}

so you must send the pricefeed address or remove it from the deploy.
```

**Reply by ghost** [reply]
sorry bro thank you the help, but i dont know quite what you mean by this. do you mean, if i have this...

constructor(address _priceFeed) public {
priceFeed = AggregatorV3Interface(_priceFeed);
owner = msg.sender;
} 

in my constructor, i must change _priceFeed to price_feed_address?

**Reply by ghost** [reply]
i fixed it by removed the price_feed_address line in FundMe.deploy. Thanks for your help

**Reply by cromewar** [reply]
That exactly what I meant, the price feed address was a constructor parameter, but you didn't have it declared on your function, I'm glad is solved now closing this thread,.
